### DISCUSSION TITLE
How can I find out if the person sending a slash command is in a channel?

### DISCUSSION THREAD

**Discussion by ThomasOToole65** [discordjs/discord.js#8822] [Q&A] [answered]
Everytime I try to use interaction.member or interaction.user I am encountering the same problem of:
`TypeError: Cannot read properties of undefined (reading 'member')`
Here is my code:

```
run: async ({ interaction, client }) =>{
        const voiceChannel = interaction.member.voice.channel
        if (!voiceChannel) {
            return interaction.editReply("You need to be in a voice channel to use this 
                                                        command");
        }
```
Any help understanding why member is undefined is much appreciated.

**Comment by almostSouji** [comment, marked as answer]
"member" is not undefined, it cannot read property `.member` OF undefined, suggesting that `interaction` has the value `undefined`.
This in turn suggests, that whatever you call `run(...)` with is not an object of the structure `{interaction: value1, client: value2}`.

If you are following a guide, you might want to re-view how they define and pass these function arguments. Please note, that both GitHub discussions and the Discord server require some understanding of JavaScript, so that you can understand the answers given to you.

You can find some common errors you may encounter during development at <https://discordjs.guide/popular-topics/errors.html#javascript-errors>
